src/1_api.js

```diff
@@ -51,12 +51,32 @@ Server.prototype.serializeObject = function(obj, prefix) {
  * @param {Object.<string, *>} data
  */
 Server.prototype.getUrl = function(resource, data) {
-	var k, v, err;
-	var url = resource.destination + resource.endpoint;
+	var k, v, err,
```

Comment:
Sounds good. I just started on it with some of the easy changes - but I can wrap that up and just push up here. Then you can take it from there.

Oh a tip, it you do any refactoring to the `XHRRequest` method - which would be another good thing to do, be very careful of the order that you set properties on the request when it comes to Internet explorer. For example, if you set `req.timeout` *before* calling `req.open`, you'll have an indecipherable error in IE 9 and 10.

We have standard error messages in `0_utils.js`, those XHR errors could certainly go in there.